metabox/metabox/core/machine.py

```diff
@@ -495,6 +504,84 @@ def get_file_transfer(self):
             file_tranfer_list.append((filename, self.dest))
         return file_tranfer_list
 
+    def get_early_dir_transfer(self):
+        provider_path = pkg_resources.resource_filename(
+            "metabox", "metabox-provider"
+        )
+        return [(provider_path, "/home/ubuntu/metabox-provider")]
+
+    def get_setup_overlay_fs(self):
+        """
+        Overlay here is used to put the metabox provider in the checkbox snap
+        This is done via a service that mounts an overlay fs on the
+        checkbox snap where we have installed the metabox provider
+        """
+        snap_runtime_name = self.CHECKBOX_CORE_SNAP_MAP[self.config.alias]
+        snap_runtime_loc = "/snap/{}/current".format(snap_runtime_name)
+
+        overlay_dir = "/home/ubuntu/.overlay"
+        work_dir = "/home/ubuntu/.work_overlay"
+        service_cmd = (
+            "/bin/mount -t overlay -o "
+            "lowerdir={snap_runtime_loc},upperdir={overlay_dir},workdir={work_dir} "
+            "overlayfs {snap_runtime_loc}"
+        ).format(
+            snap_runtime_loc=snap_runtime_loc,
+            overlay_dir=overlay_dir,
+            work_dir=work_dir,
+        )
+
+        service_content = (
+            textwrap.dedent(
+                """
+            [Unit]
```

Comment:
this (and the rest) should be indented too I think. Or is it a black thing?

Reply:
black thing I think, I can try to re-run it